source/Mission.cpp

```diff
@@ -264,6 +279,22 @@ void Mission::Load(const DataNode &node)
 
 
 
+void Mission::FinishLoading()
+{
+	for(auto &it : missionSubs)
+	{
+		if(it.second.find("assign:") != string::npos)
+			it.second=it.second.substr(7);
+		else if(it.second.find("phrase:") != string::npos)
+		{
+			const Phrase *phrase = GameData::Phrases().Get(it.second.substr(7));
+			it.second = phrase ? phrase->Get() : "";
```

Comment:
This isn't quite a flexible as you'd like. If a job uses a phrase substitution, then the phrase used will only ever differ when the game is loaded: any jobs created from the template will have the same phrase substitution during the same player session.

To fix this you need to maintain the (always non-null) Phrase pointer and invoke it in each use of Mission::Instantiate. (Which then means you do not need Mission::FinishLoading at all, since you will be storing string replacements and Phrase replacements in different maps, and don't need to prefix things with `assign:` or `phrase:`)

Reply:
I did something similar to what you said. I have to retain the order of the non-phrase substitutions, so I'm still using a vector of pairs. However, this does let me get rid of FinishLoading, and reduce the code by a dozen lines or so.